bqskit/compiler/gateset.py

```diff
@@ -231,5 +231,9 @@ def __repr__(self) -> str:
         """Detailed representation of the GateSet."""
         return self._gates.__repr__().replace('frozenset', 'GateSet')
 
+    def __hash__(self) -> int:
+        """Hash of the GateSet."""
+        return self.__repr__().__hash__()
```

Comment:
Is this hash dependent on the order in which gates were appended to the set or some other internal ordering? That is, can two equal gate sets have different hashes with this method?